tests/kernel/interrupt/src/interrupt_util.h

```diff
@@ -99,6 +99,66 @@ static inline void trigger_irq(int irq)
 	printk("Triggering irq : %d\n", irq);
 	z_arc_v2_aux_reg_write(_ARC_V2_AUX_IRQ_HINT, irq);
 }
+
+/*
+ * Trigger an interrupt of x86 need a immediate number, so hard code the
+ * irq here, for dynamic and static isr testing.
+ */
+#elif defined(CONFIG_X86)
+
+#define TEST_IRQ_0_LINE	16
+#define TEST_IRQ_1_LINE	17
+#define TEST_VECTOR_0		(TEST_IRQ_0_LINE + 32)
+#define TEST_VECTOR_1		(TEST_IRQ_1_LINE + 32)
+
+#define TEST_IRQ_DYN_LINE	18
+#define TEST_VECTOR_DYN		(TEST_IRQ_DYN_LINE + 32)
+
+static inline void trigger_irq(int irq)
+{
+	printk("Triggering irq : %d\n", irq);
+
+	/* trigger self-defined interrupt */
+	switch (irq) {
```

Comment:
This switch seems silly as the vectors are a simple function of the interrupt numbers.  Note that the other architectures you added don't try to interpret the numbers.   Why not:

```c
static inline void trigger_irq(int irq)
{
    int vector = irq + 32;
    __asm__ volatile("int %0" :: "i"(vector));
}
```

Reply:
Wow, it works. This stuck me for a long time, you just solve it in few seconds. thanks so much!